platform/android/SCsub

```diff
@@ -74,3 +74,10 @@ if lib_arch_dir != "":
         str(env["ANDROID_NDK_ROOT"]) + "/sources/cxx-stl/llvm-libc++/libs/" + lib_arch_dir + "/libc++_shared.so"
     )
     env_android.Command(out_dir + "/libc++_shared.so", stl_lib_path, Copy("$TARGET", "$SOURCE"))
+
+    out_openxr_dir = "#platform/android/java/lib/libs/openxr/" + lib_type_dir + "/" + lib_arch_dir
+    env_android.Command(
+        out_openxr_dir + "/libopenxr_loader.so",
```

Comment:
Currently we only place these into the template, but they will not be added to the APK. When an AAR with an libopenxr_loader.so is added to the project, that will be included in the final APK. 

Unfortunately, due to the Android Gradle Plugin's limitation there is no safe way telling which .so should be added to the APK if multiple files with the same name are present during build. Here is an open issue about it in Google's tracker: https://issuetracker.google.com/issues/214395989

There are 2 options to solve this:
- Package the KHR loader that we build as a standard plugin that is included with the templates, so it can be easily added to the project. This is the option I would recommend.
- Hack the Gradle scripts e.g. to detect during the final copy that the OpenXR loader is missing from the APK and patch the APK.

I would support the first option is that a lot of Android projects are not AR/VR titles, so they don't need OpenXR anyways, and this way the APK will not include the additional ~500KB loader code.

What do you think?

Reply:
I would not be against having the default Khronos loader downloadable as an AAR.

What I think is important is to make sure that we can deploy the Khronos loader, Meta loader and Pico loader together. I think this will work seeing we can turn on/off AARs in the export. This will enable people to create exports from the same project for different platforms.

Reply:
Should we generate an AAR + gdap file during the generateGodotTemplates task? If yes, can we do that in a follow-up PR, or do we need to do it in this one?

Reply:
I'm on the fence about that one, @akien-mga what is your opinion on this?

@kisg btw did you get my DM on the Godot chat server? I shared a repo with you based on your Meta AAR with CI building both the AAR and a proper asset that can be uploaded to the asset library.

We could use that same approach for the Khronos loader.

Reply:
I'm also in favor of the first option, so let's proceed with that.

> Should we generate an AAR + gdap file during the generateGodotTemplates task?

What's the behavior when no loader is provided? Is the logic smart enough to detect that and abort?

If it's, then the Khronos loader plugin should be generated separately from the Godot templates and provided as an asset as @BastiaanOlij mentioned alongside the other (e.g: Meta) loader plugins.

If the logic can't handle a missing loader, then it should be generated alongside the Godot template and provided as default.

Reply:
The first option sounds good to me too. It makes sense for both loaders to be provided as plugins, with Khronos being the one available out of the box since it's FOSS.

Reply:
@BastiaanOlij @m4gr3d @akien-mga It looks like we are in agreement here, thank you.

If you agree, we will add a new task ./gradlew generateKhronosLoaderOpenXRPlugin (or similar - please suggest a better name) that will generate the AAR and the gdap file in the bin folder.

Reply:
I would just call it `generateHKROpenXRPlugin`. I'd also like the standardise the naming of the AARs so that if you have a couple you support they're nicely together in the plugins folder. 
So I called the meta one godot_openxr_meta (godotopenxrmeta.gdap + godotopenxrmeta.aar), I suggest we would call this one godot_openxr_khr (godotopenxrkhr.gdap + godotopenxrhkr.aar).

Also right now the meta one builds a -debug.aar and -release.aar but there doesn't seem to be an option in gdap files to export either debug or release versions like we have with GDextensions. Does it make sense to even indifferentiate between the two?

Lastly, should we put all OpenXR loader plugins together in one repo and upload them as a pack to the asset library so it's easy to just install them instead of separating this out?

Reply:
@BastiaanOlij If you want to move the building of the KHR plugin to the external repo, then we should just stop building the libopenxr_loader for Android, and add the KHR plugin including the OpenXR Loader sources to the new repo. Maybe we should just use CMake (integrated with Gradle) in that new repo to build the loader so that a single Gradle invocation can build all the plugins. Maybe the in-progress godot_openxr_meta repo should be refactored to hold all the plugins? What do you think?

Regarding debug / release in .gdap: I think this would be a good feature to add, somehow like this:
```
binary=.../currentway.aar
binary_release=.../release.aar
binary_debug=.../debug.aar
```
If binary_release or binary_debug is defined, it takes precedence over binary for the respective build type, but the binary directive could still be available so existing .gdap files could still work.

@m4gr3d What do you think about this?

Reply:
> @kisg I'm also in favor of the first option, so let's proceed with that.
> 
> > Should we generate an AAR + gdap file during the generateGodotTemplates task?
> 
> What's the behavior when no loader is provided? Is the logic smart enough to detect that and abort?
> 
> If it's, then the Khronos loader plugin should be generated separately from the Godot templates and provided as an asset as @BastiaanOlij mentioned alongside the other (e.g: Meta) loader plugins.
> 
> If the logic can't handle a missing loader, then it should be generated alongside the Godot template and provided as default.

The logic can handle the missing loader, this was one of the reasons for introducing the dlopen() based approach. So we don't have to provide the loader in the default template.

Reply:
Removed the building of openxr_loader on Android

Reply:
> Regarding debug / release in .gdap: I think this would be a good feature to add, somehow like this:
> 
> ```
> binary=.../currentway.aar
> binary_release=.../release.aar
> binary_debug=.../debug.aar
> ```
> 
> If binary_release or binary_debug is defined, it takes precedence over binary for the respective build type, but the binary directive could still be available so existing .gdap files could still work.
> 
> @m4gr3d What do you think about this?

@kisg  I think that's a good idea, and it should be fairly straightforward to implement!
Let me know if you have any questions about the way to proceed.